### PR TITLE
Handler csv reader options in cudf-polars

### PR DESCRIPTION
**Pull request by wence-**
## Description

Previously we were just relying on the default cudf read_csv options which doesn't do the right thing if the user has configured things.

Now that polars passes through the information to us, we can handle things properly, and raise for unsupported cases.

While here, update to new polars release and adapt tests to bug fixes that have been made upstream.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by jameslamb — APPROVED** [review]

File: python/cudf_polars/cudf_polars/dsl/ir.py [hunk 1]
```diff
@@ -206,24 +208,104 @@ def __post_init__(self) -> None:
...
+
+            # polars skips blank lines at the beginning of the file
+            pieces = []
+            for p in self.paths:
+                skiprows = self.reader_options["skip_rows"]
+                # TODO: read_csv expands globs which we should not do,
+                # because polars will already have handled them.
+                path = Path(p)
+                with path.open() as f:
+                    while f.readline() == "\n":
```
Comment (mroeschke) [inline python/cudf_polars/cudf_polars/dsl/ir.py:287]: Is this something that `read_csv` should eventually implement?

File: python/cudf_polars/cudf_polars/dsl/ir.py [hunk 2]
```diff
@@ -206,24 +208,104 @@ def __post_init__(self) -> None:
         if self.file_options.n_rows is not None:
             raise NotImplementedError("row limit in scan")
         if self.typ not in ("csv", "parquet"):
+            raise NotImplementedError(f"Unhandled scan type: {self.typ}")
+        if self.cloud_options is not None and any(
+            self.cloud_options[k] is not None for k in ("aws", "azure", "gcp")
```
Comment (mroeschke) [inline python/cudf_polars/cudf_polars/dsl/ir.py:213]: Confirming that these strings are guaranteed to be in the dict at this point? If this option isn't supported at all would it be worthwhile just to only accept `None` here?

Comment (wence-) [reply python/cudf_polars/cudf_polars/dsl/ir.py:213]: csv reader has non-None cloud_options but the values are None, so I can't punt for `cloud_options is not None`.

File: python/cudf_polars/cudf_polars/dsl/ir.py (hunk 1 above)
Comment (wence-) [reply python/cudf_polars/cudf_polars/dsl/ir.py:287]: At some point (maybe soon?), cudf-polars is not going to use `cudf.read_csv`, but rather the libcudf layer directly, so we'll probably need to do this here anyway.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**